locust/templates/index.html

```diff
@@ -238,18 +238,46 @@ <h2>Import CSV</h2>
                         <small>Example : $.production.accounts (data belongs to accounts section, where accounts belongs to production section)</small>
                     </div>
                     <br>
-                    <div class="required_field">
-                        <label for="json_option" title="Choose desired action for current data">Action</label>
-                        <div>
-                            <input type="radio" name="json_option" id="json_option" value="replace"> replace<br>
-                            <input type="radio" name="json_option" id="json_option" value="append"> append<br>
-                        </div>
+
+                    <div>
+                        <label for="input type csv">Input type csv</label>
+                        <select id="input_type" name="input_type">
+                            <option value="all_data">All data</option>
+                            <option value="each_row">Each row per key</option>
+                        </select>
                     </div>
                     <br>
+
                     <div id="column_name_container">
                         <label for="column_name">Choose column name which has multiple data</label>
                         <ul id="column_name" class="column_name"></ul>
                     </div>
+                    <br>
+
+                    <div id="json_option_class" class="json_option_class">
+                        <label for="json_option" title="Choose desired action for current data">Action</label>
+                        <div>
+                            <input type="radio" name="json_option" id="json_option" value="replace"> Replace<br>
+                            <input type="radio" name="json_option" id="json_option" value="append"> Append<br>
+                        </div>
+                        <br>
+                    </div>
+
+                    <div id="input_method" class="input_method">
+                        <label for="input_method" title="Choose input method for each row action">Method</label>
+                        <div>
+                            <input type="radio" name="input_method_opt" id="input_method_opt" value="depend_on_key" onclick="handle_input_method(this);"> Depend on key<br>
+                            <input type="radio" name="input_method_opt" id="input_method_opt" value="in_sequence" onclick="handle_input_method(this);"> In sequence<br>
+                        </div>
+                        <br>
+                    </div>
+
+                    <div id="key_csv_json" class="key_csv_json">
+                        <label for="Key CSV">Key CSV</label><br>
+                        <input type="text" name="key_csv" id="key_csv" class="val"><br><br>
+                        <label for="Key JSON">Key JSON</label><br>
+                        <input type="text" name="key_json" id="key_json" class="val"><br><br>
```

Comment:
how about using input group showing the root path. so user knows that s/he only need to specify the last key. you can take a look at https://getbootstrap.com/docs/4.0/components/input-group/